kalendar/src/commonMain/kotlin/com/himanshoe/kalendar/KalendarAerial.kt

```diff
@@ -0,0 +1,197 @@
+        kalendarDayLabelKonfig = kalendarKonfig.kalendarDayLabelKonfig,
+        restrictToCurrentWeek = restrictToCurrentWeek,
+        events = events,
+        backgroundColor = kalendarKonfig.backgroundColor,
+        startDayOfWeek = startDayOfWeek
+    )
+}
+
+@Composable
+private fun KalendarAerialContent(
+    selectedDate: LocalDate,
+    modifier: Modifier,
+    showDayLabel: Boolean,
+    onDaySelectionAction: OnDaySelectionAction,
+    dayKonfig: KalendarDayKonfig,
+    kalendarHeaderKonfig: KalendarHeaderKonfig,
+    kalendarDayLabelKonfig: KalendarDayLabelKonfig,
+    restrictToCurrentWeek: Boolean,
+    events: KalendarEvents,
+    backgroundColor: KalendarColor,
+    startDayOfWeek: DayOfWeek
+) {
+    val today = remember { Clock.System.todayIn(TimeZone.currentSystemDefault()) }
+
+    var currentDay by remember { mutableStateOf(selectedDate) }
+    var rangeStartDate by remember { mutableStateOf<LocalDate?>(null) }
+    var rangeEndDate by remember { mutableStateOf<LocalDate?>(null) }
+    val coroutineScope = rememberCoroutineScope()
+    val selectedRange = remember { mutableStateOf<KalendarSelectedDayRange?>(null) }
+    var clickedNewDate by remember { mutableStateOf(selectedDate) }
+    val daysOfWeek = DayOfWeek.entries.rotate(distance = startDayOfWeek.ordinal)
+    val pagerState = rememberPagerState(
+        initialPage = Int.MAX_VALUE / 2,
+        pageCount = { Int.MAX_VALUE }
+    )
+    val calendarIconEnabled = pagerState.currentPage != Int.MAX_VALUE / 2
+    val headerText = remember(currentDay) {
+        getWeekDates(
+            currentDay = currentDay,
+            startDayOfWeek = startDayOfWeek
+        ).buildHeaderText()
+    }
+
+    Column(
+        modifier = modifier.background(brush = Brush.linearGradient(backgroundColor.value)),
+        horizontalAlignment = Alignment.CenterHorizontally,
+    ) {
+        KalendarHeader(
+            modifier = Modifier,
+            title = headerText,
+            arrowShown = false,
+            calendarIconEnabled = calendarIconEnabled,
+            showCalendarIcon = true,
+            onNavigateToday = {
+                if (calendarIconEnabled) {
+                    coroutineScope.launch {
+                        currentDay = today
+                        pagerState.animateScrollToPage(page = Int.MAX_VALUE / 2)
+                    }
+                }
+            },
+            kalendarHeaderKonfig = kalendarHeaderKonfig,
+            canNavigateBack = !restrictToCurrentWeek || currentDay > today,
+        )
+
+        HorizontalPager(
+            state = pagerState,
+            modifier = Modifier.fillMaxWidth()
+        ) { page ->
+            val startDate = selectedDate.plus(
+                value = (page - Int.MAX_VALUE / 2) * 7,
+                unit = DateTimeUnit.DAY
+            )
+            val displayDates = getWeekDates(
+                currentDay = startDate,
+                startDayOfWeek = startDayOfWeek
+            )
```

Comment:
need to remember this?

Reply:
I didn't see a huge difference, hence didn't do it.

But all these will be revisited later.